src/Symfony/Component/HtmlSanitizer/HtmlSanitizer.php

```diff
@@ -22,19 +23,23 @@
  */
 final class HtmlSanitizer implements HtmlSanitizerInterface
 {
-    private ParserInterface $parser;
+    private \Closure $parser;
 
     /**
      * @var array<string, DomVisitor>
      */
     private array $domVisitors = [];
 
+    /**
+     * @param \Closure(string):\Dom\Node|\DOMNode|null $parser
```

Comment:
Should you update the signature as you call the closure with 2 parameters? But `ParserInterface::parse()` only takes one.
```suggestion
     * @param \Closure(string, string):\Dom\Node|\DOMNode|null $parser
```

There would be no ambiguity with an interface.

Reply:
This doc needs to be updated.